### ISSUE TITLE
something wrong with JSON.parse

### ISSUE THREAD

**Issue by HolyWalley** [LendingHome/pipe_operator#1]
```
ruby -v
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]

irb
2.5.1 :001 > require "pipe_operator"
 => true
2.5.1 :002 > include ::PipeOperator
 => Object
2.5.1 :003 > -9.pipe { abs | Math.sqrt | to_i }
 => 3
2.5.1 : > [9, 64].map(&Math.|.sqrt)
 => [3.0, 8.0]
2.5.1 : > [9, 64].map(&Math.|.sqrt.to_i.to_s)
 => ["3", "8"]
2.5.1 : > require 'json'
 => true
2.5.1 : > "https://api.github.com/repos/ruby/ruby".pipe { URI.parse | Net::HTTP.get | JSON.parse }
Traceback (most recent call last):
       10: from /Users/yakavkrasnov/.rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        9: from (irb):7
        8: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator.rb:14:in `__pipe__'
        7: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator/pipe.rb:11:in `new'
        6: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator/pipe.rb:40:in `__call__'
        5: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator/pipe.rb:22:in `open'
        4: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator/pipe.rb:40:in `block in __call__'
        3: from /Users/yakavkrasnov/.rvm/gems/ruby-2.5.1/gems/pipe_operator-0.0.1/lib/pipe_operator/pipe.rb:40:in `instance_exec'
        2: from (irb):7:in `block in irb_binding'
        1: from /Users/yakavkrasnov/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/json/common.rb:155:in `parse'
ArgumentError (wrong number of arguments (given 0, expected 1..2))
```

But it works with 
```
"https://api.github.com/repos/ruby/ruby".pipe { URI.parse | Net::HTTP.get }
```

**Comment by shuber** [comment]
Thanks for testing @HolyWalley! Check out the `PIPE_OPERATOR_REBIND` section in https://github.com/LendingHome/pipe_operator#caveats

tldr there are two workarounds:
* `require "json"` BEFORE calling `__pipe__` for the first time
* Start up the process via `ENV["PIPE_OPERATOR_REBIND"] = "1" irb`

But note that this won't be necessary if we port this over to Ruby as a core feature!
